File: examples/python/advanced/testing-modes/mode-agnostic-tests.py
@pytest.mark.durable_execution(
    handler=my_function.handler,
    lambda_function_name="my function",
)
def test_my_function(durable_runner):
    """Test works in both local and cloud modes."""
    with durable_runner:
        result = durable_runner.run(input={"value": 42}, timeout=10)
    
    # These assertions work in both modes
    assert result.status == InvocationStatus.SUCCEEDED
    assert result.result == "expected output"

File: examples/java/advanced/testing-modes/mode-agnostic-tests.java
@Test
void testMyFunction() {
    // Works with both LocalDurableTestRunner and CloudDurableTestRunner
    var runner = LocalDurableTestRunner.create(String.class, new MyHandler());

    var result = runner.runUntilComplete(Map.of("value", 42));

    // These assertions work in both modes
    assertEquals(ExecutionStatus.SUCCEEDED, result.getStatus());
    assertEquals("expected output", result.getResult(String.class));
}
